### DISCUSSION TITLE
MYSQL fails to install due to incorrect GPG Key

### DISCUSSION THREAD

**Discussion by furquan39** [apache/airflow#36230] [General] [answered]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.4.1

### What happened?

When attempting to rebuild an apache airflow image, we are getting failures during our builds when trying to run apt-get update.

The error we see:
![image](https://github.com/apache/airflow/assets/37553943/c1c87c61-c60f-41d2-828d-3ba8734bde99)

### What you think should happen instead?

The airflow image should build successfully without any errors.

### How to reproduce

Create a Dockerfile. Add the following lines to the Dockerfile:

`FROM apache/airflow:2.4.1-python3.8

ENV SSH_PASSWD "root:Docker!"

RUN pip3 install msal Authlib flask_oauthlib azure-storage-file-datalake apache-airflow-providers-databricks==3.1.0

USER root

RUN apt-get update \
    && apt-get install -y --no-install-recommends \
    openssh-server \
    && apt-get autoremove -yqq --purge \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*`

### Operating System

Debian 11

### Deployment

Docker-Compose

### Anything else?

Seems like a similar issue to this: https://github.com/apache/airflow/issues/20911

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
I converted it to discussion thinking it was a new issue, but it seems Oracle did it again and changed the key as they did in the past - converted it back to the issue https://github.com/apache/airflow/issues/36231

**Comment by potiuk** [comment]
Wuick workaround:

```Dockerfile
USER root
RUN sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys A8D3785C
RUN  apt-get update ....
```

**Comment by potiuk** [comment]
Please let me know if that fixes your issues - will make it easier to apply a permanent fix :)

**Reply by potiuk** [reply]
cc: @furquan39 @Robso-creator @jeremiahmai @ImpressionableRaccoon @andretayer @guru-pochineni

**Reply by ghost** [reply]
hi @potiuk. 
First, thanks for the quickly response.

I'am getting the same error after added this line.
Both local and on GHA.

**Reply by potiuk** [reply]
Can you post full log please?

**Reply by furquan39** [reply]
[error]#7 0.874 Get:12 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [260 kB]
[error]#7 0.981 Get:13 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [17.7 kB]
[error]#7 1.094 Get:14 https://apt.postgresql.org/pub/repos/apt bullseye-pgdg/main amd64 Packages [305 kB]
[error]#7 2.444 W: GPG error: http://repo.mysql.com/apt/debian bullseye InRelease: The following signatures were invalid: EXPKEYSIG 467B942D3A79BD29 MySQL Release Engineering <mysql-build@oss.oracle.com>
[error]#7 2.444 E: The repository 'http://repo.mysql.com/apt/debian bullseye InRelease' is not signed.
[error]#7 ERROR: process "/bin/bash -o pipefail -o errexit -o nounset -o nolog -c apt-get update     && apt-get install -y --no-install-recommends     openssh-server     && apt-get autoremove -yqq --purge     && apt-get clean     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
[error]------
[error] > [ 3/10] RUN apt-get update     && apt-get install -y --no-install-recommends     openssh-server     && apt-get autoremove -yqq --purge     && apt-get clean     && rm -rf /var/lib/apt/lists/*:
[error]0.765 Get:8 https://packages.microsoft.com/debian/11/prod bullseye/main all Packages [1214 B]
[error]0.785 Get:9 https://packages.microsoft.com/debian/11/prod bullseye/main arm64 Packages [27.5 kB]
[error]0.794 Get:10 https://packages.microsoft.com/debian/11/prod bullseye/main armhf Packages [24.9 kB]
[error]0.795 Get:11 https://packages.microsoft.com/debian/11/prod bullseye/main amd64 Packages [131 kB]
[error]0.874 Get:12 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [260 kB]
[error]0.981 Get:13 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [17.7 kB]
[error]1.094 Get:14 https://apt.postgresql.org/pub/repos/apt bullseye-pgdg/main amd64 Packages [305 kB]
[error]2.444 W: GPG error: http://repo.mysql.com/apt/debian bullseye InRelease: The following signatures were invalid: EXPKEYSIG 467B942D3A79BD29 MySQL Release Engineering <mysql-build@oss.oracle.com>
[error]2.444 E: The repository 'http://repo.mysql.com/apt/debian bullseye InRelease' is not signed.
[error]------
[error]Dockerfile:11
[error]--------------------
[error]  10 |     
[error]  11 | >>> RUN apt-get update \
[error]  12 | >>>     && apt-get install -y --no-install-recommends \
[error]  13 | >>>     openssh-server \
[error]  14 | >>>     && apt-get autoremove -yqq --purge \
[error]  15 | >>>     && apt-get clean \
[error]  16 | >>>     && rm -rf /var/lib/apt/lists/*
[error]  17 |     
[error]--------------------
[error]ERROR: failed to solve: process "/bin/bash -o pipefail -o errexit -o nounset -o nolog -c apt-get update     && apt-get install -y --no-install-recommends     openssh-server     && apt-get autoremove -yqq --purge     && apt-get clean     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100

**Comment by potiuk** [comment, marked as answer]
Seems that we are hitting a cache problem - some people get old and some new key.

For now workaround is this:


```
RUN rm /etc/apt/sources.list.d/mysql.list
```
